pkg/proxy/config/config.go

```diff
@@ -371,3 +373,111 @@ func (c *NodeConfig) handleDeleteNode(obj interface{}) {
 		c.eventHandlers[i].OnNodeDelete(node)
 	}
 }
+
+// ServiceCIDRHandler is an abstract interface of objects which receive
+// notifications about ServiceCIDR object changes.
+type ServiceCIDRHandler interface {
+	// OnServiceCIDRAdd is called whenever creation of new ServiceCIDR object
+	// is observed.
+	OnServiceCIDRAdd(serviceCIDR *networking.ServiceCIDR)
+	// OnServiceCIDRUpdate is called whenever modification of an existing
+	// ServiceCIDR object is observed.
+	OnServiceCIDRUpdate(oldServiceCIDR, serviceCIDR *networking.ServiceCIDR)
+	// OnServiceCIDRDelete is called whenever deletion of an existing
+	// ServiceCIDR object is observed.
+	OnServiceCIDRDelete(serviceCIDR *networking.ServiceCIDR)
+	// OnServiceCIDRSynced is called once all the initial event handlers were
+	// called and the state is fully propagated to local cache.
+	OnServiceCIDRSynced()
+}
```

Comment:
ServiceCIDRs aren't going to change much, and the full list of active ServiceCIDRs is always going to be short anyway. Just simplify this to `OnServiceCIDRsChanged()`, and have the `ServiceCIDRConfig` do the work of keeping the current list, and just pass a new list down every time it changes. (And then there's only 1 dummy method to add to the backends that aren't supporting this.)

(NodeConfig should have worked this way too...)